openmls/src/treesync/node/leaf_node.rs

```diff
@@ -563,60 +602,64 @@ impl LeafNode {
     // ----- Validation ----------------------------------------------------------------------------
 
     /// Validate the leaf node in the context of a key package.
-    // TODO(#1186)
     #[allow(unused)]
-    pub(crate) fn validate_in_key_package(&self) -> Result<&Self, LeafNodeValidationError> {
+    pub(crate) fn unvalidate(self) -> LeafNode<StateUnvalidated> {
+        LeafNode {
+            payload: self.payload,
+            signature: self.signature,
+            phantom: Default::default(),
+        }
+    }
+
+    /// Validate the leaf node in the context of a key package.
+    pub(crate) fn validate_in_key_package(
```

Comment:
Yes. However, there are a few instances in the code where it happens so that we need to go from `ValidKeyPackage` to `ValidUpdate`. Maybe these can be resolved in some other way. As an alternative, we would use `invalidate().validate_in_<other>`.